fe/fe-core/src/main/java/com/starrocks/load/loadv2/BrokerLoadJob.java

```diff
@@ -331,6 +334,71 @@ private void createLoadingTask(Database db, BrokerPendingTaskAttachment attachme
         }
     }
 
+    @Override
+    public void afterAborted(TransactionState txnState, boolean txnOperated, String txnStatusChangeReason)
+            throws UserException {
+        if (!txnOperated) {
+            return;
+        }
+        writeLock();
+        try {
+            // check if job has been completed
+            if (isTxnDone()) {
+                LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                        .add("state", state)
+                        .add("error_msg", "this task will be ignored when job is: " + state)
+                        .build());
+                return;
+            }
+            if (retryTime <= 0 || !txnStatusChangeReason.contains("timeout")) {
```

Comment:
How about using LoadJob.getDeadlineMs() to know whether the job is timeout.

Reply:
fixed